packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1391,22 +1391,60 @@ export class TransactionController extends BaseController<
     return rawTransactions;
   }
 
+  /**
+   * Updates the transaction hash associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param hash - The new hash value to be assigned.
+   */
+  updateTransactionHash(transactionId: string, hash: string) {
+    const transactionMeta = this.getTransaction(transactionId);
+    if (!transactionMeta) {
+      log(`Transaction with id ${transactionId} not found.`);
+      return;
+    }
+    transactionMeta.hash = hash;
+    this.updateTransaction(
+      transactionMeta,
+      'TransactionController:updateTransactionHash - Transaction hash updated',
+    );
+  }
+
+  /**
+   * Updates the transaction status associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param status - The new status value to be assigned.
+   */
+  updateTransactionStatus(transactionId: string, status: TransactionStatus) {
```

Comment:
As mentioned in the extension PR, should we use this PR to introduce a `updateCustodialTransaction` method that would limit updates to those with `custodyId` and also only allow updating the subset of properties needed by MMI?

Reply:
yes, I created `updateCustodialTransaction` checking `custodyId` and only allowed to update `status` (failed, submitted and signed), hash and status.